clouddriver-kubernetes-v2/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesCoreCachingAgentTest.java

```diff
@@ -446,4 +437,124 @@ public void wrongNamespace(int numCachingAgents) {
           .asMap();
     }
   }
+
+  /**
+   * Given a collection of OnDemandAgent.OnDemandResult, return all cache results in these on-demand
+   * results.
+   */
+  private static ImmutableMap<String, Collection<CacheData>> extractCacheResults(
+      Collection<CacheResult> onDemandResults) {
+    return onDemandResults.stream()
+        .map(result -> result.getCacheResults().entrySet())
+        .flatMap(Collection::stream)
+        .collect(
+            ImmutableSetMultimap.flatteningToImmutableSetMultimap(
+                Map.Entry::getKey, e -> e.getValue().stream()))
+        .asMap();
+  }
+
+  /** Given a collection of ProviderCache, return all on-demand entries in these caches. */
+  private static ImmutableMap<String, Collection<CacheData>> extractCacheEntries(
+      Collection<ProviderCache> providerCaches) {
+    return providerCaches.stream()
+        .map(providerCache -> providerCache.getAll("onDemand"))
+        .flatMap(Collection::stream)
+        .filter(Objects::nonNull)
+        .map(
+            cacheData -> {
+              try {
+                return objectMapper.<Map<String, Collection<CacheData>>>readValue(
+                    (String) cacheData.getAttributes().get("cacheResults"),
+                    new TypeReference<Map<String, Collection<DefaultCacheData>>>() {});
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            })
+        .map(Map::entrySet)
+        .flatMap(Collection::stream)
+        .collect(
+            ImmutableSetMultimap.flatteningToImmutableSetMultimap(
+                Map.Entry::getKey, e -> e.getValue().stream()))
+        .asMap();
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {1, 2, 10})
+  public void loadData(int numAgents) {
+    String deploymentKey =
+        Keys.InfrastructureCacheKey.createKey(
+            KubernetesKind.DEPLOYMENT, ACCOUNT, NAMESPACE1, DEPLOYMENT_NAME);
+
+    String storageClassKey =
+        Keys.InfrastructureCacheKey.createKey(
+            KubernetesKind.STORAGE_CLASS, ACCOUNT, "", STORAGE_CLASS_NAME);
+
+    ImmutableCollection<KubernetesCoreCachingAgent> cachingAgents =
+        createCachingAgents(getNamedAccountCredentials(), numAgents);
+    LoadDataResult loadDataResult = processLoadData(cachingAgents, ImmutableMap.of());
+
+    assertThat(loadDataResult.getResults()).containsKey(DEPLOYMENT_KIND);
+    assertThat(loadDataResult.getResults().get(DEPLOYMENT_KIND))
+        .extracting(data -> data.getAttributes().get("name"))
+        .containsExactly(DEPLOYMENT_NAME);
+
+    assertThat(loadDataResult.getResults()).containsKey(STORAGE_CLASS_KIND);
+    assertThat(loadDataResult.getResults().get(STORAGE_CLASS_KIND))
+        .extracting(data -> data.getAttributes().get("name"))
+        .containsExactly(STORAGE_CLASS_NAME);
+
+    assertThat(loadDataResult.getResults()).containsKey(DEPLOYMENT_KIND);
```

Comment:
Is there a reason we have this assertion twice (once on line 496 and once here)?

Reply:
Good catch!  I had intended one set of assertions to check that the return value of `loadData` had the expected entries, and the other set to check that the cache actually contained the expected entries, but was checking the return value in both.

When trying to add the assertion on the contents of the cache, I realized that `loadData` doesn't actually update the cache itself---the return value is used by another function to actually update the cache.  So I just removed the duplicate set of assertions, as updating the cache is not part of `loadData`'s contract.